testdata/default/cheats/StringUtils.t.sol

```diff
@@ -39,4 +39,16 @@ contract StringManipulationTest is DSTest {
         assertEq("World", splitResult[1]);
         assertEq("Reth", splitResult[2]);
     }
+
+    function testIndexOf() public {
```

Comment:
Other tests I'd recommend:
- If key is empty it reverts (assuming this is expected behavior)
- `vm.indexOf("", nonEmptyFuzzedKey) == type(uint256).max`

Reply:
empty key `"anything".find("")` always returns `0` in Rust, I don't think any input should revert

Reply:
That works for me

Reply:
> Other tests I'd recommend:
> 
> * If key is empty it reverts (assuming this is expected behavior)
> * `vm.indexOf("", nonEmptyFuzzedKey) == type(uint256).max`

It will return 0 in case of empty string, which is also the case with [solady](https://github.com/Vectorized/solady/blob/b61da94a2d275471eec03b584d70ff31881ecfae/src/utils/LibString.sol#L551-L554). lmk wdyt.

Reply:
I can the empty string test case and also add a note saying it will return 0 in case of empty string.